File: patterns/c-patterns/pattern29.c
// Pattern 29: ...

//  * 1 * 2 *
//  3 * 4 * 5
//  * 6 * 7 *
//  8 * 9 *10
//  *11 *12 *

#include <stdio.h>
int main()
{
    int n = 5;
    int i, j;
    int x = 1;

    for (i = 1; i <= n; i++)
    {
        for (j = 1; j <= n; j++)
        {
            if ((i + j) % 2 == 1)
            {
                printf("%2d", x++);
            }
            else
            {
                printf("%2c", '*');
            }
        }
        printf("\n");
    }
    return 0;
}

File: patterns/javascript-patterns/pattern29.js
// Pattern 29: ...

//  * 1 * 2 *
//  3 * 4 * 5
//  * 6 * 7 *
//  8 * 9 *10
//  *11 *12 *

const n = 5;
let x = 1;

for (let i = 1; i <= n; i++) {
    let row = '';

    for (let j = 1; j <= n; j++) {
        if ((i + j) % 2 === 1) {
            row += `${x++} `;
        } else {
            row += '* ';
        }
    }

    console.log(row);
}
